core/src/main/java/hudson/model/RunParameterDefinition.java

```diff
@@ -151,8 +152,8 @@ public ParameterDefinition newInstance(StaplerRequest req, JSONObject formData)
             return req.bindJSON(RunParameterDefinition.class, formData);
         }
         
-        public AutoCompletionCandidates doAutoCompleteProjectName(@QueryParameter String value) {
-            return AutoCompletionCandidates.ofJobNames(Job.class, value, null, Jenkins.getInstance());
+        public AutoCompletionCandidates doAutoCompleteProjectName(@QueryParameter String value, @AncestorInPath Item self, @AncestorInPath ItemGroup container) {
```

Comment:
Ideally it needs to be annotated with `@Restricted(NoExternalUse.class)` in order to prevent binary compatibility concerns. In this case I do not see external usages, so it should be fine